Simplified/Book/UI/NYPLBookCellDelegate+Audiobooks.m

```diff
@@ -25,48 +25,63 @@ @implementation NYPLBookCellDelegate (Audiobooks)
 #pragma mark - Audiobook Methods
 
 - (void)openAudiobook:(NYPLBook *)book {
-  NSURL *const url = [[NYPLMyBooksDownloadCenter sharedDownloadCenter] fileURLForBookIndentifier:book.identifier];
-  NSData *const data = [NSData dataWithContentsOfURL:url];
-  if (data == nil) {
-    [self presentCorruptedItemErrorWithLog:@{
-      @"book": book.loggableDictionary ?: @"N/A",
-      @"fileURL": url ?: @"N/A"
-    }];
+  // If an audiobook download is in progress, we should use the existing AudiobookManager.
+  DefaultAudiobookManager *audiobookManager = [[NYPLMyBooksDownloadCenter sharedDownloadCenter] audiobookManagerForBookID:book.identifier];
+  if (audiobookManager) {
+    [self presentAudiobook:book withAudiobookManager:audiobookManager];
     return;
   }
+  
+  [AudiobookManifestAdapter transformAudiobookManifestWithBook:book
+                                                    completion:^(NSDictionary<NSString *,id> * _Nullable json,
+                                                                 id<DRMDecryptor> _Nullable decryptor,
+                                                                 enum AudiobookManifestError error) {
+    if (error == AudiobookManifestErrorCorrupted) {
```

Comment:
since `error` is actually an `enum` it would be more solid if we handled this with a `switch` because then we can list all the possible cases, and the compiler can tell us if we missed one.